File: Programmers/javascript/hanghae99_4th/q40_신규_아이디_추천.js
function solution(new_id) {
    var answer = new_id
        .toLowerCase() // 1단계
        .replace(/[^\w\-\.]/g, '') // 2단계
        .replace(/[\.]{2,}/g, '.') // 3단계
        .replace(/^\./, '') // 4단계
        .replace(/\.$/, ''); // 4단계
    if (answer.length == 0)
        // 5단계
        answer = 'a';
    if (answer.length >= 16)
        // 6단계
        answer = answer.slice(0, 15).replace(/\.$/, '');
    if (answer.length <= 2)
        // 7단계
        answer = answer.padEnd(3, answer[answer.length - 1]);
    return answer;
}

console.log(solution('...!@BaT#*..y.abcdefghijklm'));

File: Programmers/python/prac/q40_신규_아이디_추천.py
import re

def solution(new_id):
    answer = re.sub('\.$', '', re.sub('^\.', '', re.sub('[\.]{2,}', '.', re.sub('[^\w\-\.]', '', new_id.lower()))))

    answer = 'a' if len(answer) == 0 else answer[0:15]
    answer = re.sub('\.$', '', answer)
    if len(answer) <= 2:
        answer += "".join([answer[len(answer)-1] for i in range(3-len(answer))])
    return answer
